src/Nethermind/Nethermind.Merge.Plugin/MergeGossipPolicy.cs

```diff
@@ -24,23 +24,50 @@ namespace Nethermind.Merge.Plugin
 {
     public class MergeGossipPolicy : IGossipPolicy
     {
-        private readonly IGossipPolicy _preMergeGossipPolicy;
-        private readonly IPoSSwitcher _poSSwitcher;
         private readonly IManualBlockFinalizationManager _blockFinalizationManager;
 
         public MergeGossipPolicy(
             IGossipPolicy? apiGossipPolicy, 
-            IPoSSwitcher? poSSwitcher, 
             IManualBlockFinalizationManager blockFinalizationManager)
         {
-            _preMergeGossipPolicy = apiGossipPolicy ?? throw new ArgumentNullException(nameof(apiGossipPolicy));
-            _poSSwitcher = poSSwitcher ?? throw new ArgumentNullException(nameof(poSSwitcher));
+            ShouldGossipBlocks = apiGossipPolicy?.ShouldGossipBlocks ?? throw new ArgumentNullException(nameof(apiGossipPolicy));
             _blockFinalizationManager = blockFinalizationManager;
+            
+            _blockFinalizationManager.BlocksFinalized += OnBlockFinalized;
         }
 
-        public bool ShouldGossipBlocks => !_poSSwitcher.HasEverReachedTerminalBlock() && _preMergeGossipPolicy.ShouldGossipBlocks;
+        // According to spec (https://github.com/ethereum/EIPs/blob/d896145678bd65d3eafd8749690c1b5228875c39/EIPS/eip-3675.md#network)
```

Comment:
Not finalized but terminal

The networking stack SHOULD NOT send the following messages if they advertise the descendant of any terminal PoW block:

[NewBlockHashes (0x01)](https://github.com/ethereum/devp2p/blob/master/caps/eth.md#newblockhashes-0x01)
[NewBlock (0x07)](https://github.com/ethereum/devp2p/blob/master/caps/eth.md#newblock-0x07)